cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -425,6 +443,34 @@ impl LockedManifestCatalog {
         map.into_values()
     }
 
+    fn split_fully_locked_groups(
+        groups: impl IntoIterator<Item = PackageGroup>,
+        seed_lockfile: Option<&LockedManifestCatalog>,
+    ) -> (Vec<LockedPackageCatalog>, Vec<PackageGroup>) {
+        let seed_locked_packages = seed_lockfile.map_or_else(HashMap::new, Self::make_seed_mapping);
+
+        let (already_locked_groups, groups_to_lock): (Vec<_>, Vec<_>) =
+            groups.into_iter().partition(|f| {
+                f.descriptors
+                    .iter()
+                    .all(|descriptor| descriptor.derivation.is_some())
+            });
+
+        // convert already locked groups back to locked packages
+        let already_locked_packages = already_locked_groups
+            .iter()
+            .flat_map(|group| {
+                group.descriptors.iter().flat_map(|d| {
+                    seed_locked_packages
+                        .get(&(&d.install_id, &group.system))
+                        .map(|(_, locked)| LockedPackageCatalog::clone(locked))
+                })
+            })
+            .collect::<Vec<_>>();
+
+        (already_locked_packages, groups_to_lock)
+    }
```

Comment:
the original implementation in pkgdb was able to group the descriptors first _by group_ and then compare the groups.
Creating groups of `(group, system) x (install_id, descriptor, locked_package)` and then mapping these groups into a `(Vec<locked_package>, Vec<PackageGroup>)` all in one makes that function rather complex.